### PR TITLE
add getOrd to Array and relevant tests (Fixes 354)

### PR DESCRIPTION
**Pull request by sledorze**

### CODE REVIEW

File: src/Array.ts [hunk 1]
```diff
@@ -42,6 +44,27 @@ export const getMonoid = <A = never>(): Monoid<Array<A>> => {
   }
 }
 
+export const getSetoid: <A>(S: Setoid<A>) => Setoid<A[]> = getArraySetoid
+
+export const getOrd = <T>(O: Ord<T>): Ord<Array<T>> => ({
```
Comment (gcanti) [inline src/Array.ts]: Just a nitpick: maybe we can rename the type parameter `T` to `A` (for consistency)

File: src/Array.ts [hunk 2]
```diff
@@ -42,6 +44,27 @@ export const getMonoid = <A = never>(): Monoid<Array<A>> => {
...
+  ...getSetoid(O),
+  compare: (a: Array<T>, b: Array<T>): Ordering => {
```
Comment (gcanti) [inline src/Array.ts]: Also here

File: src/Array.ts [hunk 3]
```diff
@@ -42,6 +44,27 @@ export const getMonoid = <A = never>(): Monoid<Array<A>> => {
...
+    const lengthOrder = ordNumber.compare(a.length, b.length)
+    if (lengthOrder !== 0) {
+      return lengthOrder
+    } else {
+      const len = a.length
```
Comment (gcanti) [inline src/Array.ts]: This can be moved up since is used in `ordNumber.compare(a.length, b.length)`

**Comment by gcanti** [comment]
> the definition of Ord for foldable leads to the same output for any two arrays

what do you mean?

**Comment by sledorze** [comment]
@gcanti I've checked the purescript implementation - for reference:
https://github.com/purescript/purescript-prelude/blob/master/src/Data/Ord.js

I propose to make the necessary changes to match this behaviour.
The drawback is that it is less efficient in case of array of different length (no shortcut).

@gcanti do you agree on this?

**Comment by gcanti** [comment]
@sledorze I see, that's another possible order for arrays. I'm ok with that, better to add some documentation on how the algorithm works though

File: src/Array.ts [hunk 4]
```diff
@@ -42,6 +44,24 @@ export const getMonoid = <A = never>(): Monoid<Array<A>> => {
   }
 }
 
+export const getSetoid: <A>(S: Setoid<A>) => Setoid<A[]> = getArraySetoid
```
Comment (gcanti) [inline src/Array.ts]: `@function` missing

File: src/Array.ts [hunk 5]
```diff
@@ -42,6 +44,24 @@ export const getMonoid = <A = never>(): Monoid<Array<A>> => {
...
+
+export const getOrd = <A>(O: Ord<A>): Ord<Array<A>> => ({
```
Comment (gcanti) [inline src/Array.ts]: - `@function` missing
- could you please add some docs?

**Pull request merged**